runestone/mchoice/test/test_assess.py

```diff
@@ -63,12 +62,6 @@ class MultipleChoiceQuestion_Tests(RunestoneTestCase):
     def test_ma1(self):
         """Multiple Answer: Nothing selected, Check button clicked"""
         self.driver.get(self.host + "/index.html")
-        wait = WebDriverWait(self.driver, 10)
```

Comment:
I'm curious as to why you removed this block.  Our tests are riddled with the kinds of timing problems that this is supposed to fix.

Reply:
I added a [default implicit wait of 5 seconds](https://github.com/RunestoneInteractive/RunestoneComponents/pull/1105/files#diff-80bc6a407ddd8b12c6f879353243ef7e16fd852f16d0ad34a06975d28f1272f6R230), which should be fine for most cases. Therefore, we can remove most of the explicit waits, which are rather ugly code-wise.

Reply:
I think that works for some things but not everything.  Particularly when we are getting the text from an element and then using `self.assertIn("foo", element.text)`  See https://github.com/RunestoneInteractive/RunestoneComponents/commit/4074b39202123749462f380c9c0bc71edbeb83be for my updates to the activecode tests that should fix that in a foolproof way.

Reply:
I agree -- more specific conditions can/should be added.  I think with the implicit 5 second wait, you just need to assert that the expected condition is met, so:

```
       wait = WebDriverWait(self.driver, 10)
        wait.until(
            EC.text_to_be_present_in_element((By.ID, "test1_stdout"), "Hello")
        )
```

can (I think) become

`assert(EC.text_to_be_present_in_element((By.ID, "test1_stdout"), "Hello"))`

per the docs on [explicit waits](https://selenium-python.readthedocs.io/waits.html#explicit-waits).

Reply:
That does not seem to work for me.  If I change hello to "sdlkfjsh" the assert still works.  even though the only text in the element is "Hello World". Whereas with the wait.until if I look for the "sdlkfjsh" it times out and fails.

Reply:
A slightly cleaner solution for the text fields is to create a `self.wait` object in setUp.  Then we can do:

```
        self.wait.until(
            EC.text_to_be_present_in_element((By.ID, "test1_stdout"), "Hello World"),
            message="Did not find expected text",
        )
```
If we are only looking for one thing in the output then there is no need for a separate assert as it will cause the test to error if the text is not there after 5 seconds.

If you want to look for multiple things in the output then you can wait for the first one and use simple asserts for the rest.

Reply:
OK, thanks for trying. I like your idea of creating a `wait` object in `setUp` -- that seems like a good way to go.